### PR TITLE
Release 2.7.0

### PR DESCRIPTION
**Pull request by AlvinSchiller**
> [!CAUTION]
> Do NOT use the latest RaspberryPi OS Bookworm release (also don't upgrade via apt) with the new kernel 6.6, if you need to use GPIO components . There have been incompatible changes for the RPi.GPIO library, so Buttons, Shims and Rfid Reader usage can be broken ([#2295](https://github.com/MiczFlor/RPi-Jukebox-RFID/discussions/2295)).
>**Using RaspberryPi OS _Bullseye_ is the current recommendation as it does not use the new kernel.**

### Features
- **Support for Spotify again!**
  - a prerelease of mopidy-spotify is used to bring back the spotify integration. Please see [Spotify-FAQ](https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Spotify-FAQ) for further information
  - this includes startup speed improvements for mopidy due to playlists refresh optimization in the prerelease
- Raspberry Pi OS **Buster** and Python < 3.9 will not be supported/tested anymore with this release.
  - official Buster support is running out by June 30th, 2024 
  - mopidy-spotify minimum support is Bullseye

### Improvements
- some refactorings of the installation script
- better installation logging
- documentation updates
- version updates for github actions

### Fixes
- fix for special characters entered during setup (spotify credentials, wifi, ...)
- fix for function calls from RotaryEncoder
- update defaults and fixed handling for some yes/no questions 
- improved startup time for wifi in combination with the autohotspot feature

### Note
Version 3 is becoming mature and will soon be the new default of Phoniebox. Therefore Version 2 is slowly going into a maintenance mode and no new features will be added to this version.

### CODE REVIEW

**Comment by AlvinSchiller** [comment]
Would like to also integrate https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2357

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        python-version: ['3.9', '3.10', '3.11', '3.12']
```
Comment (s-martin) [inline .github/workflows/pythonpackage.yml:18]: We are also dropping 3.8 here (release note state only dropping 3.7)

**Review by s-martin — APPROVED** [review]
Massive startup speed improvements especially for Spotify could also be mentioned in the release notes.

File: settings/version-number [hunk 2]
```diff
@@ -1 +1 @@
-2.6.0
+2.7.0-alpha
```
Comment (Groovylein) [inline settings/version-number:1]: Should be changed?

**Review by Groovylein — APPROVED** [review]

File: settings/version-number (hunk 2 above)
Comment (AlvinSchiller) [reply settings/version-number:1]: Yep, will be the last commit before the merge, as the ci will fail after it ^^

**Pull request merged**